SU-HSDO/suhumsci
# SHS-4860: js to capture search terms and expand timeline/accord

## Summary
This will append a user's search query to their search result links. When they click on a search result, if the page contains collapsed accordion or timeline components containing that search phrase, it will automatically expand them. 

## Steps to Test
1. Go to an accordion component page (`/components/accordions-secondary`) and a timeline component page (`/components/vertical-timeline-without-and-collection`).
2. Edit those pages and within a couple of the components, add a unique search term. I added 'Pepperoni pizza'. Make sure to include the term in a collection or well for testing.
3. Reindex your local search.
4. Use the search toolbar in the header to search for the term you just inserted.
5. Click on the search results. Notice how the accordion/timeline components that contain your search are automatically expanded. Click expand/collapse buttons to make sure original functionality is intact.
6. Visit a regular page and visit the accordion/timeline pages **without** the appended query params in the URL. Check the console to ensure their are no errors. 

## Screenshots
![Screenshot 2023-04-12 at 3 06 09 PM](https://user-images.githubusercontent.com/8405274/231561498-5b8df93e-7408-4f2a-a807-9bcaee785f06.png)

## Notes
* There is no new twig template. There's so hair preprocessing going on in the background again so the theme suggestion for creating a views search is not working. There exists a pre-existing twig template that's used for searches, so I included the JS there. I 'if'ed the function in case this template is mysteriously used elsewhere. I cannot tell.
* I know there's been a lot of discussion about refining the release standards, so I'm not sure if there's a '4k release branch' anymore.
